### ISSUE TITLE
Select item with TAB

### ISSUE THREAD

**Issue by metalfox6383** [Pixabay/jQuery-autoComplete#29]
Hi:
It would be useful to launch onSelect with Tab. I added the condition at the line 123:

else if (e.which == 13 || e.which == 9) {

I hope that helps. Until now I don't have problems.

Greetings.

**Comment by SimonSteinberger** [comment]
I'm not sure if onSelect should be called here. Basically, then it should also be called when browsing through suggestions with up/down cursor keys and then mouse clicking somewhere else on the page. It's essentially the same result ...

**Comment by metalfox6383** [comment]
Hi. You right. It happens that I need launch onSelect when the user press TAB. Many users don't have the habit of pressing ENTER after select the item. Of those users, some press TAB for change the control focus.

**Comment by SimonSteinberger** [comment]
You can bind any keypress event manually to the input field and check for the tab key.

**Comment by metalfox6383** [comment]
How? I was try, but any result. Obviously with the wrong way. How I call onSelect from the keyUp event?

**Comment by SimonSteinberger** [comment]
onSelect is your own function and you can read the input's current value directly. So, unless you need some extra data that is attached to the suggestions (which is possibly only accessibly via the called onSelect), it should work.

In order to have the onSelect function called by the plugin, it may be possible to manually dispatch the enter key as event on the input field - when pressing tab ... just a thought.

**Comment by LexRiver** [comment]
yes, i think onSelect must be called on Tab key, because we can change (select) the value with Tab key. Else value must not be changed.

**Comment by LexRiver** [comment]
that works for me:
`
                // enter or tab
                else if (e.which == 13 || e.which == 9) {
                    if(that.sc.is(':visible')) {
                        var sel = $('.autocomplete-suggestion.selected', that.sc);
                        if (sel.length) {
                            o.onSelect(e, sel.data('val'), sel);
                            setTimeout(function() { that.sc.hide(); }, 10);
                        }
                    }
                }
`

**Comment by SimonSteinberger** [comment]
I guess it should be an option, because there are different use-cases. Sometimes the "tab select" is desired, sometimes not. I've put it on my to-do list.

**Comment by metalfox6383** [comment]
Hi. Yes... I work with LexRiver`s solution, but I had a problem when the focus pass across the elements, the "onSelect" event fire. I need trigger onSelect only when I chose a element.
I resolved my problem adding a new data variable called "autocomplete_in_process" to TRUE or FALSE when the list of suggestion show and hide respectively.
But I insist that the event should be triggered with the TAB key, too. Greetings. The component is wonderful.
